dlt/_workspace/runtime.py

```diff
@@ -0,0 +1,191 @@
+import os
+from typing import Optional
+
+from dataclasses import dataclass
+import jwt
+from git import Union
+from tomlkit.toml_file import TOMLFile
+
+from dlt._workspace._workspace_context import WorkspaceRunContext
+from dlt._workspace.configuration import WorkspaceRuntimeConfiguration
+from dlt._workspace.exceptions import (
+    LocalWorkspaceIdNotSet,
+    RuntimeNotAuthenticated,
+    RuntimeOperationNotAuthorized,
+    WorkspaceIdMismatch,
+    WorkspaceRunContextNotAvailable,
+)
+from dlt._workspace.runtime_clients.api.api.me import me
+from dlt._workspace.runtime_clients.api.client import Client as ApiClient
+from dlt._workspace.runtime_clients.api.models.me_response import MeResponse
+from dlt._workspace.runtime_clients.auth.client import Client as AuthClient
+from dlt._workspace.cli.config_toml_writer import WritableConfigValue, write_values
+from dlt.common.configuration.providers.toml import (
+    SECRETS_TOML,
+    ConfigTomlProvider,
+    SecretsTomlProvider,
+)
+from dlt.common.configuration.resolve import resolve_configuration
+from dlt.common.configuration.specs.pluggable_run_context import RunContextBase
+from dlt.common.runtime.run_context import active
+
+
+@dataclass
+class AuthInfo:
+    user_id: str
+    email: str
+    jwt_token: str
+
+
+class RuntimeAuthService:
+    """
+    Implements login, logout and auth check internals
+
+    Authentication is performed based on the JWT token stored in the global secrets. On top of that,
+    authorization uses organisation and workspace id stored in the local config. For that, depending on the usage,
+    either workspace run context or base run context is required.
+    """
+
+    auth_info: Optional[AuthInfo] = None
+
+    _run_context: RunContextBase
+    _local_workspace_id: Optional[str] = None
+    _remote_workspace_id: Optional[str] = None
+
+    def __init__(self, run_context: RunContextBase):
+        self._run_context = run_context
+
+    @property
+    def workspace_run_context(self) -> WorkspaceRunContext:
```

Comment:
most of the methods below shoudl be private. or not?

Reply:
please see main comment